dvc/output/local.py

```diff
@@ -18,6 +18,12 @@ class OutputLOCAL(OutputBase):
     REMOTE = RemoteLOCAL
     sep = os.sep
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        if self.repo and path_isin(self.def_path, self.repo.root_dir):
+            self.def_path = relpath(self.def_path, self.repo.root_dir)
```

Comment:
Ok, we are missing one more case here. def_path is the path that is written into the dvcfile, which is not guaranteed to be in the repo root, so we need to relpath against `self.stage.wdir`.

Reply:
Presumably the previous line (`path_isin`) also would need updating then?

Reply:
@casperdcl Not necessarily. You could actually use `if self.repo and self.is_in_repo` instead even :slightly_smiling_face:

Reply:
I'm not sure I follow the logic of the proposed change though, which is:

```python
if def_path is in root:
  def_path = relative to wdir  # this assumes def_path is in wdir, which was not tested
```

Reply:
@casperdcl def_path is resolved relative to `stage.wdir` when it is loaded, so it has to be relative to `wdir` not the repo root.

Reply:
my concern is this:


```python
if path_isin('/c/c.dat', '/c/'):
    print(os.path.relpath('/c/c.dat', '/d/wdir/'))
'../../c/c.dat'
```

Reply:
@casperdcl Could you elaborate why it is a concern for you, please?

Reply:
well the last line (what will appear in the DVC-file) should be `/c/c.dat` without the `../..` prefix. If `def_path` is resolved relative to `stage.wdir` then we need:

```diff
- if def_path is in root:
+ if def_path is in wdir:
    def_path = relative to wdir
```

Reply:
ok after discussion I think relative to `wdir` (rather than `root`) is fine. Not sue about the `self.is_in_repo` since that uses `os.path.isabs` which will abvs fail in this case